sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -341,10 +366,34 @@ pi_result cuda_piEventRetain(pi_event event);
 
 /// \endcond
 
-_pi_event::_pi_event(pi_command_type type, pi_context context, pi_queue queue)
+_pi_queue::native_type _pi_queue::get_compute() {
+  if (num_compute_streams_ < compute_streams_.size()) {
+    unsigned int idx = num_compute_streams_++;
+    if (idx < compute_streams_.size()) {
+      PI_CHECK_ERROR(cuStreamCreate(&compute_streams_[idx], flags_));
+    }
+  }
+  return compute_streams_[compute_stream_idx_++ % compute_streams_.size()];
```

Comment:
You are right that this could be an issue, but even `std::call_once` does not resolve it completely. Streams are also accessed by `for_each_stream`, so we need a mutex.